hack/python-sdk/gen-sdk.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubeflow Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SWAGGER_JAR_URL="https://repo1.maven.org/maven2/org/openapitools/openapi-generator-cli/5.1.0/openapi-generator-cli-5.1.0.jar"
+SWAGGER_CODEGEN_JAR="hack/python-sdk/openapi-generator-cli.jar"
+SWAGGER_CODEGEN_CONF="hack/python-sdk/swagger_config.json"
+SWAGGER_CODEGEN_FILE="pkg/apis/kubeflow/v1/swagger.json"
+SDK_OUTPUT_PATH="sdk/python"
+
+if [ -z "${GOPATH:-}" ]; then
+    export GOPATH=$(go env GOPATH)
+fi
+
+# Backup existing openapi_generated.go
+mv pkg/apis/kubeflow/v1/openapi_generated.go pkg/apis/kubeflow/v1/openapi_generated.go.backup
+
+echo "Generating OpenAPI specification ..."
+openapi-gen --input-dirs github.com/kubeflow/mpi-operator/pkg/apis/kubeflow/v1,github.com/kubeflow/common/pkg/apis/common/v1 --output-package github.com/kubeflow/mpi-operator/pkg/apis/kubeflow/v1 --go-header-file hack/boilerplate/boilerplate.go.txt
+
+echo "Generating swagger file ..."
+go run hack/python-sdk/main.go 0.1 > ${SWAGGER_CODEGEN_FILE}
```

Comment:
Do we want to sync SDK version with MPI Operator release ?
WDYT @gaocegege @zw0610 @terrytangyuan ?

Reply:
Because the python sdk only relies on the mpijob api definition, I believe we don't need to sync sdk version with the mpi-operator version. Instead, maybe I should change it to the api version (`V1`).

Reply:
What if we decide to add some new APIs to the MPIJob `v1` ?
In that case, what version we should add to the MPI SDK ?

Reply:
@andreyvelich WDYT?

Reply:
Sorry for the long reply.
I think we can follow the Kubernetes API structure where each new field is added to `v1alpha1 -> v1beta1 -> v1`.
I am not sure how we will do it in the MPIJob.

@terrytangyuan @gaocegege What do you think ?

Reply:
Yes we've completed phases from v1alpha1 to v1 so we should probably bump to v2alpha soon.

Reply:
@terrytangyuan Do you have any ideas how we should increase the SDK version in that case ?
Should it point to the MPI operator release version or how @zw0610 proposes here: https://github.com/kubeflow/mpi-operator/pull/357#discussion_r628672538 ?

Reply:
Yes I think what's proposed in https://github.com/kubeflow/mpi-operator/pull/357#discussion_r628672538 sounds good to me.

Reply:
Ok, then lgtm from my side. Thank you for doing this @zw0610!